### PR TITLE
Add Geo Location posture check

### PR DESCRIPTION
**Pull request by bcmmbaga**
## Describe your changes

This PR extends the existing `/api/posture-checks` endpoint with  Geo location check.

### Checklist
- [ ] Is it a bug fix
- [ ] Is a typo/documentation fix
- [x] Is a feature enhancement
- [ ] It is a refactor
- [ ] Created tests that fail without the change (if possible)
- [ ] Extended the README / documentation, if necessary

### CODE REVIEW

File: management/server/http/posture_checks_handler.go [hunk 1]
```diff
@@ -230,6 +235,25 @@ func validatePostureChecksUpdate(req api.PostureCheckUpdate) error {
...
+	if geoLocationCheck := req.Checks.GeoLocationCheck; geoLocationCheck != nil {
+		if geoLocationCheck.Action == "" {
+			return status.Errorf(status.InvalidArgument, "action for geolocation check shouldn't be empty")
+		}
+		if len(geoLocationCheck.Locations) == 0 {
+			return status.Errorf(status.InvalidArgument, "locations for geolocation check shouldn't be empty")
+		}
+
+		for _, loc := range geoLocationCheck.Locations {
+			if loc.CountryCode == "" {
```
Comment (surik) [inline management/server/http/posture_checks_handler.go:252]: Maybe we can add basic validation to check the `2-letter ISO 3166-1 alpha-2 code that represents the country`. Later we can validate against location DB.

File: management/server/posture/geo_location.go [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+	// Locations list of geolocations, to which the policy applies
+	Locations []Location
+
+	// Action to take upon policy match
+	Action string
+}
+
+func (g *GeoLocationCheck) Check(peer nbpeer.Peer) (bool, error) {
+	for _, loc := range g.Locations {
+		if loc.CountryCode == peer.Meta.Location.CountryCode && loc.CityName == peer.Meta.Location.CityName {
```
Comment (surik) [inline management/server/posture/geo_location.go]: We need to consider county-wide checks when CityName is empty

**Review by surik — CHANGES_REQUESTED** [review]

File: management/server/peer/peer.go [hunk 3]
```diff
@@ -66,6 +66,12 @@ type PeerSystemMeta struct {
...
 	KernelVersion string
+	// Location mock location for peer
+	// TODO: Add actual implementation based on peer IP
+	Location struct {
+		CountryCode string
+		CityName    string
+	} `gorm:"embedded;embeddedPrefix:location_"`
 }
 
 func (p PeerSystemMeta) isEqual(other PeerSystemMeta) bool {
```
Comment (surik) [inline management/server/peer/peer.go:77]: We probably need to extend this with Location too

Comment (bcmmbaga) [reply management/server/peer/peer.go:77]: I left it intentionally, as we should cover this when we extend the peer metadata with location

File: management/server/http/posture_checks_handler.go [hunk 4]
```diff
@@ -230,6 +236,27 @@ func validatePostureChecksUpdate(req api.PostureCheckUpdate) error {
...
+				return status.Errorf(status.InvalidArgument, "country code for geolocation check shouldn't be empty")
+			}
+
+			countryCodeRegex := regexp.MustCompile("^[a-zA-Z]{2}$")
```
Comment (surik) [inline management/server/http/posture_checks_handler.go]: We can initialise this as a global variable. No need to compile it each time.

**Review by surik — APPROVED** [review]

**Pull request merged**